tracer/src/Datadog.Trace/AppSec/Waf/Context.cs

```diff
@@ -110,6 +104,25 @@ public void CommitUserRuns(Dictionary<string, object> addresses, bool fromSdk)
         }
     }
 
+    private bool ShouldRunWith(IDatadogSecurity security, UserEventsState.UserRecord? userRecord, string? value, string address, bool fromSdk)
+    {
+        if (value is null || !security.AddressEnabled(address))
+        {
+            return false;
+        }
+
+        string? previousValue = null;
+        var previousValueFromSdk = false;
+        if (userRecord.HasValue)
```

Comment:
If userRecord.HasValue is false, could we just return fromSdk and avoid the string comparison?

Reply:
Very good point, changed in [c9a71ab](https://github.com/DataDog/dd-trace-dotnet/pull/6623/commits/c9a71ab22fabea013d8b322de76671623a2be800)